Randomize the list of team members on the people page

bsweger opened the pull request:
This is a low-friction way to shuffle the list of team members on [https://reichlab.io/people](https://reichlab.io/people)

The trade-off is that the order will change when the site is rebuilt but not when the page is refreshed.

If we want the lab's leaders to always be at the top, that should be do-able with a few more lines of code.

bsweger committed:
Randomize the list of team members on the people page

This change shuffles the array of team members read from team.yml.
The order will update every time the site is rebuilt, but it won't
update when the page is refreshed.

bsweger commented:
Here's what it looks like on localhost: 

![image](https://github.com/reichlab/reichlab.github.io/assets/540544/80862f8e-3179-4902-8ef4-b7344f18dd6a)

And after rebuilding the site:

![image](https://github.com/reichlab/reichlab.github.io/assets/540544/2699f318-d723-4ccb-a16d-47f27696bb0f)

nickreich reviewed (changes requested):
Amazingly "low-friction"! Nice.

I think we should do this. My one comment is that maybe we should add a small piece of text just below the "People" header and before the "person cards" start that says something about the ordering, so visitors know what is going on. Something like "The ordering of lab members below is randomly shuffled once every day."

Also, this doesn't impact the ordering of alumni, right?

bsweger committed:
Add a note to people.html about the randominzed order of team members

bsweger commented:
This change will also randomize the alumni list, because it randomizes everyone in team.yml. The distinction between current and former people is made downstream in people.html: https://github.com/reichlab/reichlab.github.io/blob/source/people.html#L57

If we don't want to randomize alumni, I can split the list earlier on and only shuffle the current team members.

Added some text--let me know if you'd like different wording:

![image](https://github.com/reichlab/reichlab.github.io/assets/540544/043be3be-7b1a-412e-b757-c49e78fff58b)

nickreich commented:
Maybe I'm being too finnicky here, but I actually do like the idea of keeping alumni list fixed (there is a clear linear timeline/ordering for them) and only randomizing current members.

nickreich commented:
All of this is making me wish we had netlify builds for site previews.

nickreich reviewed (changes requested):
I vote for randomizing only current members, and leaving the alumni ordered, to enable easier temporal indexing of recent grads.

bsweger committed:
Parse team.yml into current and alumni arrays

Creating distinct arrays from the first means we can remove the "type
== 'Alumni'" checks in the people.html template, and also allows us
to shuffle the order of current team members while retaining the order
of alumni.

bsweger commented:
@nickreich just pushed a change that should limit randomization to current team members.

The diff is easier to grok if you disregard whitespaces: https://github.com/reichlab/reichlab.github.io/pull/174/files?w=1

Agree about a site preview (and will create an issue and let you prioritize accordingly). In the meantime, feel free to stop by for an in-person preview (I know, I know, not the best)
